runtime/src/snapshot_utils.rs

```diff
@@ -2965,6 +2958,21 @@ pub fn purge_old_bank_snapshots(
     do_purge(bank_snapshots);
 }
 
+/// Purges all `bank_snapshots`
+///
+/// Does not exit early if there is an error while purging a bank snapshot.
+fn _purge_bank_snapshots<'a>(bank_snapshots: impl IntoIterator<Item = &'a BankSnapshotInfo>) {
+    bank_snapshots.into_iter().for_each(|bank_snapshot| {
+        let result = purge_bank_snapshot(&bank_snapshot.snapshot_dir);
+        if result.is_err() {
+            warn!(
+                "Failed to purge bank snapshot: {}",
+                bank_snapshot.snapshot_dir.display()
+            );
+        }
+    });
```

Comment:
Personal preference, but I almost never use the `for_each` iterator chain as I think a direct for loop is typically more readable/familiar.


We could do something like this (passes clippy & fmt):
```suggestion
    for snapshot_dir in bank_snapshots.into_iter().map(|s| &s.snapshot_dir) {
        if purge_bank_snapshot(snapshot_dir).is_err() {
            warn!("Failed to purge bank snapshot: {}", snapshot_dir.display());
        }
    }
```